HACKING.md

```diff
@@ -141,17 +141,82 @@ appropriately.  To install `godep` locally run:
 
 If you are not updating packages you should not need godep installed.
 
-## Updating Godeps from upstream
+## Updating Kubernetes from upstream
+
+There are a few steps involved in rebasing Origin to a new version of Kubernetes. We need to make sure
+that not only the Kubernetes packages were updated correctly into `Godeps`, but also that *all tests are
+still running without errors* and *code changes, refactorings or the inclusion/removal of attributes
+were properly reflected* in the Origin codebase. 
+
+To start, we need to rebase Origin to a new version of Kubernetes. Check out the desired branch of
+Kubernetes in your GOPATH and then run `godep restore ./...` from the Kubernetes directory. Switch to 
+the OpenShift directory and run `godep save ./...`. Example:
+
+1. `$ cd $GOPATH/src/github.com/GoogleCloudPlatform/kubernetes`
+2. `$ git checkout master`
+3. `$ git pull`
+4. `$ godep restore ./...`. This will install the package versions specified in the `Godeps/Godeps.json` 
+of Kubernetes to your GOPATH.
+5. `$ cd $GOPATH/src/github.com/openshift/origin`
+6. `$ godep save ./...`. This will save a list of the checked-out dependencies to the file 
+`Godeps/Godeps.json`, and copy their source code into `Godeps/_workspace`.
+7. If in the previous step godep complaints about the checked out revision of a package being different 
+than the wanted revision, this probably means there are new packages in Kubernetes that we need to add. 
+Do a `godep save <pkgname>` and then `$ godep save ./...` again.
+8. Read over the changes with `git status` and make sure it looks reasonable. The whole Godeps directory 
+will be added to version control, including `_workspace`.
+9. Commit using the message `bump(github.com/GoogleCloudPlatform/kubernetes):<commit SHA>`, where 
+`<commit SHA>` is the commit id for the Kubernetes version we are including in our Godeps. It can be 
+found in our `Godeps/Godeps.json` in the declaration of any Kubernetes package.
+
+Eventually during the development cycle we introduce changes to dependencies right in the Origin 
```

Comment:
I'd propose to put this part under separate header. The first part explains how to do full k8s update, whereas this describes cherry-picking just single commits. Both parts makes sense and look ok, but are disjoint IMHO.